### ISSUE TITLE
Files mandatory for a plugin type

### ISSUE THREAD

**Issue by alexandru-elisei** [mudrd8mz/moodle-tool_pluginskel#33]
A plugin is expected to have a set of required files specific to that plugin type. For example, an activity module is required to have the files lib.php, mod_form.php, index.php and view.php (and db/install.xml, but that is better generated by the XMLDB editor).

These files should be generated implicitely based on the plugin type, and the user shouldn't have to explicitely specify them in the recipe file.

For this purpose I propose that the function manager::prepare_file_skeletons() will call two separate functions:

- prepare_feature_skeletons(), which will add the file skeletons for the features requested in the recipe, in the same way that prepare_file_skeletons() does now.

- prepare_plugin_specific_skeletons(), which will add the file skeletons for the files mandatory for the plugin type.

For now, I propose that prepare_plugin_specific_skeletons() will add the plugin-specific files in series of if statements based on the plugin type:
```php
public function prepare_plugin_specific_skeletons() {
    
    $plugintype = $this->recipe['component_type'];

    if ($plugintype === 'mod') {
        $this->prepare_file_skeleton('mod_form.php', 'php_internal_file', 'mod_form');
        $this->prepare_file_skeleton('view.php', 'php_web_file', 'view');
        ...
    }

    if ($plugintype === 'block') {
         $this->prepare_file_skeleton($this->recipe['component'].'.php', 'php_internal_file', 'block');
         ...
    }
    ...
}
```

**Comment by alexandru-elisei** [comment]
It is worth noting that as we add support for generating more and more plugin types the above function can get unmanageably long. If that is the case, one solution could be to have separate functions that add the files required for each plugin type:
```php
public function prepare_plugin_specific_skeletons() {

    $plugintype = $this->recipe['component_type'];

     switch ($plugintype) {
         case: 'mod':
              $this->prepare_mod_file_skeletons();
              break;
         case: 'block':
              $this->prepare_block_file_skeletons();
              break;
         ...
         default:
               throw new exception('Unsupported plugin type: '.$plugintype);
     }
}

**Comment by alexandru-elisei** [comment]
Closing the issue, solved by pull request #45
